src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java

```diff
@@ -88,53 +93,74 @@ public String getTransactionIdForConsentRequest(String consentRequestId) {
+                            patientHealthInfoStatuses.add(statusBuilder
+                                    .status(getStatusAgainstDate(
+                                            dataRequestDetail.getConsentRequestedAt(),
+                                            serviceProperties.getConsentArtefactWaitTime()))
+                                    .build());
+                            return;
+                        }
+
+                        if (Objects.isNull(dataRequestDetail.getDataPartStatus())) {
+                            patientHealthInfoStatuses.add(statusBuilder
+                                    .status(getStatusAgainstDate(
+                                            dataRequestDetail.getDataFlowRequestedAt(),
+                                            serviceProperties.getDataPartWaitTime()))
+                                    .build());
                             return;
                         }
-                        statusesByConsentReqId.put(requestMapping.getConsentRequestId(), patientHealthInfoStatus);
+
+                        patientHealthInfoStatuses.add(statusBuilder.status(getStatusFor(dataRequestDetails)).build());
                     });
 
-                    var consentRequestIds = List.copyOf(statusesByConsentReqId.keySet());
-                    return dataFlowRepository.fetchDataPartDetails(consentRequestIds)
-                            .collectList()
-                            .filter(dataParts -> isValidRequester(dataParts, requesterId))
-                            .switchIfEmpty(Mono.error(unauthorizedRequester()))
-                            .flatMapMany(dataParts -> calculateDataPartStatuses(statusesByConsentReqId, dataParts))
-                            .mergeWith(Flux.fromIterable(patientHealthInfoStatuses));
+                    return Flux.fromIterable(patientHealthInfoStatuses);
                 });
-
     }
 
-    private Flux<PatientHealthInfoStatus> calculateDataPartStatuses(HashMap<String, PatientHealthInfoStatus> statusesByConsentReqId,
-                                                                    List<DataPartDetail> dataParts) {
-        var dataPartStatuses = new ArrayList<PatientHealthInfoStatus>();
-        Map<String, List<DataPartDetail>> dataPartsByConsentReqId = dataParts.stream()
-                .collect(Collectors.groupingBy(DataPartDetail::getConsentRequestId));
-
-        statusesByConsentReqId.forEach((consentReqId, patientHealthInfoStatus) -> {
-            var currentDataParts = dataPartsByConsentReqId.get(consentReqId);
-            var dataPartsStatus = currentDataParts == null ? DataRequestStatus.PROCESSING : getStatusFor(currentDataParts);
-            dataPartStatuses.add(patientHealthInfoStatus.toBuilder().status(dataPartsStatus).build());
-        });
+    //TODO: If someone knows a better way to do it please update this.
+    private boolean isUUID(String maybeUUID){
```

Comment:
do we care whether UUID or not?

Reply:
Because if do the database query on a non-uuid string then it fails the entire query, so we decided that any requestId that is invalid should be ignored and status for valid request ids should be responded back.